### PR TITLE
Ncnews5

### PR DESCRIPTION
**Pull request by Nat2llaw**
added error sad path and error paths

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -36,17 +44,82 @@ describe("GET/api/articles/:article_id", () => {
...
+              body: expect.any(String),
+              created_at: expect.any(String),
+              votes: expect.any(Number)
+            })
+          );
+        });
       });
   });
-  test("404:article id not in database", () => {
+  test("400:article id not in database", () => {
```
Comment (Rosem10) [inline __tests__/app.test.js:64]: Article id not in database should be a 404 not found! A 400 should be used for an article id that's of the wrong type.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -36,17 +44,82 @@ describe("GET/api/articles/:article_id", () => {
...
     return request(app)
       .get("/api/articles/1123")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Id not found" );
+      });
+  });
+});
+
+describe("GET/api/users/", () => {
```
Comment (Rosem10) [inline __tests__/app.test.js:74]: This isn't part of ticket 5.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -36,17 +44,82 @@ describe("GET/api/articles/:article_id", () => {
...
+            expect.objectContaining({
+              username: expect.any(String),
+              name: expect.any(String),
+              avatar_url: expect.any(String)
+            })
+          );
+        });
+      });
+  });
+  test("404: mispelt endpoint", () => {
```
Comment (Rosem10) [inline __tests__/app.test.js:92]: This only needs to be tested once for the whole app. It will cover any endpoints that we have not designed a path for. Therefore, your test on line 32 will cover the same thing (although in that test you should be checking that the user receives a response with the expected message, not just an empty body).

File: __tests__/app.test.js [hunk 4]
```diff
@@ -36,17 +44,82 @@ describe("GET/api/articles/:article_id", () => {
...
-        .then(({ body }) => {
-          console.log(body)
-        expect.objectContaining(body);
+      .then(({ misspeltUsers }) => {
+        expect(misspeltUsers).toEqual(undefined);
       });
   });
 });
+
+// describe.only("GET/api/articles/:article_id", () => {
```
Comment (Rosem10) [inline __tests__/app.test.js:102]: I'm not sure what ticket this PR covers? You seem to have skipped patchArticlesById and gone straight to getUsers. That's not a problem in this case, although you shouldn't push any code that isn't in use / you don't want to be merged into main. Therefore, please remove this commented-out block before merging.

Comment (Rosem10) [reply __tests__/app.test.js:102]: If this is to cover patchArticleById then you _really_ need to reinstate this test and add some error handling.

File: db/controller/controller.js [hunk 5]
```diff
@@ -1,24 +1,47 @@
...
+  fetchUsers,
+  updateVotes,
+} = require("../model/model");
 
 exports.getTopics = (req, res, next) => {
-    fetchTopics().then((topics) => {
-        res.status(200).send(topics)
+  fetchTopics()
+    .then((topic) => {
+      res.status(200).send(topic);
```
Comment (Rosem10) [inline db/controller/controller.js:11]: This should be sent back on an object please!

File: db/model/model.js [hunk 6]
```diff
@@ -9,6 +9,27 @@ exports.fetchTopics = () => {
...
+exports.fetchUsers = () => {
+  return db
+    .query(`SELECT * FROM users`)
+    .then(({ rows }) => {
+      return rows;
+    });
+};
+
+exports.updateVotes = (id, voteCount) => {
+    console.log(id)
```
Comment (Rosem10) [inline db/model/model.js:28]: Remove console.logs please!

**Review by Rosem10 — CHANGES_REQUESTED** [review]
This needs some cleaning up before it can be merged, please. Specifically, we need testing for the patch for success and errors.
